src/components/JSONEdit.tsx

```diff
@@ -46,6 +46,7 @@ const JSONEdit: React.FC<JSONEditProps & TextareaHTMLAttributes<HTMLTextAreaElem
     }
 
     updateValue(updatedData);
+    name && validateField(name);
```

Comment:
@sergei-maertens we are hitting an issue here and don't know how to get things working:

`name && validateField(name)` was added because the `updateValue` call above did not trigger the `validate` callback [here](https://github.com/open-formulieren/formio-builder/pull/69/files#diff-916b5a36551dfa9c19ff511ea1397be46e02d9e474a85121f968c4bd7c132581R42) (even though `validateOnChange` is set to `true`).

The issue is data gets out of sync, as here `updatedData` corresponds to the latest version of the user input data, whereas adding a `console.log` in the parent `validate` callback shows an outdated value.

I've tried adding a call to the base Formik `onChange`:

```ts
const {onChange: formikOnChange} = getFieldProps(name);
// Then calling `formikOnChange` in our custom `onChange`
```

But nothing works out

Reply:
I still don't think that wiring up this validate behaviour directly in the component is the right approach. Especially because this JSONEdit component is also used to edit the entire form component, but it could also point at a different subtree that has nothing to do with JSONLogic.

IMO hooking up the validator belongs in the zod validation schema, where you can make use of `.refine` or `.superRefine` to invoke this validator:

```tsx
const schema = z.object({
  itemsExpression: z.record(z.unknown()).superRefine(
    ...  // in the callback here you'd call the `infer` validator/checker
  ),
});
```

Reply:
This is cleaner indeed. Unfortunately I seem to be facing https://github.com/robertLichtnow/zod-formik-adapter/issues/15, will have to look closer.

Reply:
That seems weird since the (super)refine is used in the columns component validation.

Reply:
I see it being called with some logs, but the error doesn't seem to be displayed.